grpc/binlog/v1alpha/binarylog.proto

```diff
@@ -54,29 +78,25 @@ message GrpcLogEntry {
   // The logger uses one of the following fields to record the payload,
   // according to the type of the log entry.
   oneof payload {
```

Comment:
Option C is to encode the message type within this oneof, and leave out the extra enum:

```
oneof payload {
   Metadata request_header = 4;
   Metadata response_header = 5;
   Message request_message = 6;
   ...
   bool half_close = ...
   ...
}
```

I remember I brought this up in an earlier review on this proto definition, but I don't recall the response.  Does this make real-world usage worse?